### PR TITLE
Improve UI for selecting metrics filter

### LINKED ISSUE #1260: Metrics: Improve UI for selecting metrics filter
The current filter tag selector is a multi-select autocomplete.

Some problems:
* The dropdown closes when a tag is selected/unselected. Making multiple changes is clunky.
* There isn't a way to quickly select all tags.
* If new data with new tags becomes available while the page is open, the data with new tags aren't included in the chart until the page is refreshed. It's confusing that new data isn't immediately visible despite the user taking no action to filter.
* The browser's autocomplete displays over the top of the control's dropdown. The conflict is ugly and confusing.
* Many or long selected values adds a hard to use scrolling system to the autocomplete content.

My proposed fix is:
* `FluentOverflow` + `FluentBadge` for selected items
* Clicking on row displays a `FluentPopover` + all checkbox + list of checkboxes for tags.

### LINKED ISSUE #1287: Metrics: Add translations for view count UI
See https://github.com/dotnet/aspire/pull/1265#discussion_r1418269250

There are two pieces of text:

* `Options` header
* `View counts` label

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/dotnet/aspire/issues/1260
Fixes https://github.com/dotnet/aspire/issues/1287

* Replace `FluentAutoComplete` with badges + overflow + button + popup + checkboxes
* New dimensions are automatically selected
* New values on existing dimensions that have all selected are also selected

**Old:**

metrics-old

**New:**

metrics-new

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/1327)

### CODE REVIEW

**Review by drewnoakes — APPROVED** [review]

**Comment by adamint** [comment]
Why is this an improvement? Is this because of aesthetic concerns?

> badges + overflow + button + popup + checkboxes

Would appear to be less informative from a non-visual user, and it the explicit text box input makes it more clear that+how these are editable

**Comment by JamesNK** [comment]
The original issue details what is wrong with using an autocomplete: https://github.com/dotnet/aspire/issues/1260

> Some problems:
> * The dropdown closes when a tag is selected/unselected. Making multiple changes is clunky.
> * There isn't a way to quickly select all tags.
> * If new data with new tags becomes available while the page is open, the data with new tags aren't included in the chart until the page is refreshed. It's confusing that new data isn't immediately visible despite the user taking no action to filter.
> * The browser's autocomplete displays over the top of the control's dropdown. The conflict is ugly and confusing.
> * Many or long selected values adds a hard to use scrolling system to the autocomplete content.

File: src/Aspire.Dashboard/Components/Controls/ChartDimension.razor [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        </ChildContent>
+        <MoreButtonTemplate>
+            <FluentBadge>
+                @($"+{context.ItemsOverflow.Count()}")
+            </FluentBadge>
+        </MoreButtonTemplate>
+    </FluentOverflow>
+</div>
+<div>
+    <FluentButton id="@($"typeFilterButton-{SanitizedHtmlId}")"
```
Comment (adamint) [inline src/Aspire.Dashboard/Components/Controls/ChartDimension.razor:34]: Please add a button aria name.

**Review by adamint — CHANGES_REQUESTED** [review]

**Comment by adamint** [comment]
For accessibility reasons, I have a feeling this should be a FluentDataGrid. We have data that's very much so "griddy" - items in rows with structured columns.

**Comment by JamesNK** [comment]
> For accessibility reasons, I have a feeling this should be a FluentDataGrid. We have data that's very much so "griddy" - items in rows with structured columns.

Done

File: src/Aspire.Dashboard/Model/CounterChartViewModel.cs [hunk 2]
```diff
@@ -14,18 +14,51 @@ public class CounterChartViewModel
 [DebuggerDisplay("{DebuggerToString(),nq}")]
 public class DimensionFilterViewModel
 {
+    private string? _sanitizedHtmlId;
+
     public required string Name { get; init; }
     public List<DimensionValueViewModel> Values { get; } = new();
-    public IEnumerable<DimensionValueViewModel> SelectedValues { get; set; } = Array.Empty<DimensionValueViewModel>();
+    public HashSet<DimensionValueViewModel> SelectedValues { get; } = new();
```
Comment (adamint) [inline src/Aspire.Dashboard/Model/CounterChartViewModel.cs:21]: nit: I think this can be ISet

**Review by adamint — APPROVED** [review]

**Pull request merged**